README.md

```diff
@@ -1,11 +1,131 @@
 # irctest
 
-This is the integration test suite for Oragono, ultimately derived from [ProgVal/irctest](https://github.com/ProgVal/irctest), which is a general-purpose IRC protocol compatibility testing suite.
+This project aims at testing interoperability of software using the
+IRC protocol, by running them against test suites and making different
+software communicate with each other.
 
-Some of these tests may be applicable to other projects (we attempt to mark the tests that are only applicable to Oragono).
+It is very young and does not contain a lot of test cases yet.
 
-This suite needs more test cases. Contributions are welcome and are a great way to help the Oragono project!
+## The big picture
 
-## Running Tests
+This project contains:
+
+* IRC protocol test cases
+* small wrappers around existing software to run tests on them
+
+Wrappers run software in temporary directories, so running `irctest` should
+have no side effect, with [the exception of Sopel](https://github.com/sopel-irc/sopel/issues/946).
+
+## Prerequisites
+
+Install irctest and dependencies:
+
+```
+git clone https://github.com/ProgVal/irctest.git
+cd irctest
+pip3 install --user -r requirements.txt pyxmpp2-scram
+python3 setup.py install --user
+```
+
+Add `~/.local/bin/` to your `PATH` if it is not.
+
+```
+export PATH=$HOME/.local/bin/:$PATH
+```
+
+## Test selection
+
+irctest is invoked using the pytest test runner / CLI.
+
+A major feature of pytest that irctest heavily relies on is test selection.
+Using the `-k` option, you can select and deselect tests based on their names
+and/or markers (listed in `pytest.ini`).
+For example, you can run `LUSERS`-related tests with `-k lusers`.
+Or only tests based on RFC1459 with `-k rfc1459`.
+
+By default, all tests run; even niche ones. So you probably always want to
+use these options: `-k 'not Oragono and not deprecated and not strict`.
+This excludes:
+
+* `Oragono`-specific tests (included as Oragono uses irctest as its official
+  integration test suite)
+* tests for deprecated specifications, such as the IRCv3 METADATA
+  specification
+* tests that check for a strict interpretation of a specification, when
+  the specification is ambiguous.
+
+## Run tests
+
+To run (server) tests on Oragono:
+
+```
+cd /tmp/
+git clone https://github.com/oragono/oragono.git
+cd oragono/
+make build
+export PATH=/tmp/oragono:$PATH
+cd ~/irctest
+pytest --controller irctest.controllers.oragono -k 'not deprecated'
```

Comment:
As suggested on the other review, is `python3 -m pytest` preferable here?

Reply:
I prefer `pytest` when humans type it, because it's much shorter to type.

But I can mention at the beginning that the name may vary

Reply:
Ah, I figured it out: Ubuntu Focal provides `pytest-3` but not `pytest`. I could probably put logic in the Makefile to pick the right one.

Reply:
yes, indeed. just use `python3 -m pytest` in scripts, it works everywhere